<a href="https://colab.research.google.com/github/Kwangbin/statistic/blob/main/%EA%B0%80%EC%84%A4%EA%B2%80%EC%A0%95.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
cd drive/MyDrive/ml/statistic/data/

/content/drive/MyDrive/ml/statistic/data


In [2]:
ls

cars.csv          ch11_potato.csv        ch1_sport_test.csv  Untitled.ipynb
ch10_access.csv   ch11_training_ind.csv  ch2_scores_em.csv
ch10_enquete.csv  ch11_training_rel.csv  ch3_anscombe.npy
ch11_ad.csv       ch12_scores_reg.csv    ch4_scores400.csv


감자튀김 정량 130G

14개 표본 128.45G

우연인가? 아닌가?

가설 검증 

In [3]:
import numpy as np
import pandas as pd
from scipy import  stats

In [4]:
# 소숫점 3자리까지 표현
%precision 3
np.random.seed(1111)

In [5]:
#감자튀김 자료 불러오기
df = pd.read_csv('ch11_potato.csv')

In [6]:
df

무게
0   122.02
1   131.73
2   130.60
3   131.82
4   132.05
5   126.12
6   124.43
7   132.89
8   122.79
9   129.95
10  126.14
11  134.45
12  127.64
13  125.68

In [14]:
df.describe()

무게
count   14.000000
mean   128.450714
std      3.983585
min    122.020000
25%    125.790000
50%    128.795000
75%    131.797500
max    134.450000

In [7]:
sample = np.array(df['무게'])
sample

array([122.02, 131.73, 130.6 , 131.82, 132.05, 126.12, 124.43, 132.89,
       122.79, 129.95, 126.14, 134.45, 127.64, 125.68])

In [11]:
s_mean = np.mean(sample)
s_mean

128.451

In [15]:
s_var=np.var(sample)
s_var

14.735

정규분포를 따른다

분산은 9 이다 가정


In [16]:
# norm 정규분포  9/4  분산/표분수
rv = stats.norm(130, np.sqrt(9/14))
rv

In [19]:
# 표본 평균이 95%(0.95)아래구간   5% 이하의 확률로 128.681G 이하의 무게가 된다는 것을 알수 있다
# 즉, 128.681G  이하로 나올 활률은 5% 이다
rv.isf(0.95)

128.681

표본 평균  128.451

5% 이하의 확률 값 128.681

결국 128.681 이하의 값이 측정 되어서 우연이 아니라 의도적으로 양을 적게 해서 판매 하였다고 할수 있다

신뢰구간 밖의 값이 측정 되었다

##사장님 무게 줄이지 말고 똑바로 파세요


귀무가설, 대립가설

귀무가설  사장을 130G으로 상품을 판매 한다

대립가설 사장은 130G 이하로 물건을 판매 한다

95% 신뢰구간으로 128.681 이하면 귀무 가설을 기각 한다

##P-value 
P값이 유의수준보다 왼쪽(작다)이냐, 오른쪽(크다)이냐에 따라 귀무가설 채택 ,기각 선택

왼쪽 기각, 오른쪽 선택

##검정통계량 이 임계값보다 작을때 귀무가설 기각

In [21]:
#검정 통계량
z=(s_mean-130)/np.sqrt(9/14)
z

-1.932

In [22]:
#임계값
rv=stats.norm()
rv.isf(0.95)

-1.645

검정통계 -1.932 < 임계값 -1.645

검정통계량이 임계값보다 작다 

즉 귀무가설 기각

# 사장님 장사 똑바로 하세요

##P값을 이용한 가설 점검

유의수준 5% , 즉 0.05 보다 작으면 귀무가설 기각


In [23]:
# 검정통계량 Z를 이용해 P값 구하기
rv.cdf(z)

0.027

P 값 0.027 < 유의수준 0.05  

P값이 유의수준보다 작아서 귀무가설 기각